.github/workflows/build.yml

```diff
@@ -1,18 +1,39 @@
-name: build-demo
-on: [ push ]
+name: SonarCloud
+on:
+  push:
+    branches:
+      - main
```

Comment:
It would be best to have 2 separate workflows: one that runs the tests on each commit, and another one that is for Sonar analys on pull request.